[cgal] build failure with latest version

**Host Environment**
- OS: macOS 11.1
- Compiler: Apple clang version 12.0.0 (clang-1200.0.32.28)

**To Reproduce**
Steps to reproduce the behavior:
`./vcpkg update`
`./vcpkg upgrade --no-dry-run`

**Failure logs** 
```
Starting package 66/69: gmp:x64-osx
Building package gmp[core]:x64-osx...
CMake Error at ports/gmp/portfile.cmake:2 (message):
  Can't build gmp if mpir is installed.  Please remove mpir, and try install
  gmp again if you need it.
Call Stack (most recent call first):
  scripts/ports.cmake:128 (include)


Error: Building package gmp:x64-osx failed with: BUILD_FAILED
Please ensure you're using the latest portfiles with `./vcpkg update`, then
submit an issue at https://github.com/Microsoft/vcpkg/issues including:
  Package: gmp:x64-osx
  Vcpkg version: 2021-01-13-unknownhash

Additionally, attach any relevant sections from the log files above.
```
**Additional context**
The upgrade reinstalls boost which reinstalls cgal, which fails due to the above. Explictly attempting to install CGAL yields:
```
adam@hapkido ~/vcpkg (master) $ ./vcpkg install cgal
Computing installation plan...
The following packages will be built and installed:
    cgal[core]:x64-osx -> 5.2
  * gmp[core]:x64-osx -> 6.2.1#3
  * mpfr[core]:x64-osx -> 4.1.0
Additional packages (*) will be modified to complete this operation.
Detecting compiler hash for triplet x64-osx...
Could not locate cached archive: /Users/adam/.cache/vcpkg/archives/0e/0e4b12f212b200f36191a8ccd16474ade56aceaa.zip
Could not locate cached archive: /Users/adam/.cache/vcpkg/archives/2e/2e21dd1d6bfb1cd9bf00bd6bd2895992a18ec243.zip
Could not locate cached archive: /Users/adam/.cache/vcpkg/archives/4e/4ee8b1f3a9c6fc5040f92e822a68c6fd1cefaa70.zip
Starting package 1/3: gmp:x64-osx
Building package gmp[core]:x64-osx...
CMake Error at ports/gmp/portfile.cmake:2 (message):
  Can't build gmp if mpir is installed.  Please remove mpir, and try install
  gmp again if you need it.
Call Stack (most recent call first):
  scripts/ports.cmake:128 (include)


Error: Building package gmp:x64-osx failed with: BUILD_FAILED
Please ensure you're using the latest portfiles with `./vcpkg update`, then
submit an issue at https://github.com/Microsoft/vcpkg/issues including:
  Package: gmp:x64-osx
  Vcpkg version: 2021-01-13-unknownhash

Additionally, attach any relevant sections from the log files above.
```

Note that doing:

```
adam@hapkido ~/vcpkg (1d866090f) $ git checkout b7c8a30
Previous HEAD position was 1d866090f Add dummy version.txt for vcpkg older than vcpkg_minimum_required. (#15828)
HEAD is now at b7c8a3097 [libssh] Add support for Android (#15687)
```
Yields a successful install, so it was broken sometime after January 22nd.
```
adam@hapkido ~/vcpkg (b7c8a3097) $ ./vcpkg install cgal
Computing installation plan...
The following packages will be built and installed:
    cgal[core]:x64-osx -> 5.2
  * gmp[core]:x64-osx -> 6.2.1#3
  * mpfr[core]:x64-osx -> 4.1.0
Additional packages (*) will be modified to complete this operation.
Detecting compiler hash for triplet x64-osx...
Could not locate cached archive: /Users/adam/.cache/vcpkg/archives/0e/0e4b12f212b200f36191a8ccd16474ade56aceaa.zip
Could not locate cached archive: /Users/adam/.cache/vcpkg/archives/2e/2e21dd1d6bfb1cd9bf00bd6bd2895992a18ec243.zip
Could not locate cached archive: /Users/adam/.cache/vcpkg/archives/4e/4ee8b1f3a9c6fc5040f92e822a68c6fd1cefaa70.zip
Starting package 1/3: gmp:x64-osx
Building package gmp[core]:x64-osx...
CMake Error at ports/gmp/portfile.cmake:2 (message):
  Can't build gmp if mpir is installed.  Please remove mpir, and try install
  gmp again if you need it.
Call Stack (most recent call first):
  scripts/ports.cmake:128 (include)


Error: Building package gmp:x64-osx failed with: BUILD_FAILED
Please ensure you're using the latest portfiles with `./vcpkg update`, then
submit an issue at https://github.com/Microsoft/vcpkg/issues including:
  Package: gmp:x64-osx
  Vcpkg version: 2021-01-13-unknownhash

Additionally, attach any relevant sections from the log files above.
adam@hapkido ~/vcpkg (b7c8a3097) $ git checkout e4ce66e                  
Previous HEAD position was b7c8a3097 [libssh] Add support for Android (#15687)
HEAD is now at e4ce66eec [vcpkg] Improve diagnostics around compiler tracking and toolset errors. 'Fixes' #15823. (#15825)
adam@hapkido ~/vcpkg (e4ce66eec) $ ./vcpkg install cgal
Computing installation plan...
The following packages will be built and installed:
    cgal[core]:x64-osx -> 5.2
  * mpfr[core]:x64-osx -> 4.0.2-2
Additional packages (*) will be modified to complete this operation.
Detecting compiler hash for triplet x64-osx...
Could not locate cached archive: /Users/adam/.cache/vcpkg/archives/b2/b21576f062032d14e5e5805b229a057f2c271d00.zip
Could not locate cached archive: /Users/adam/.cache/vcpkg/archives/8e/8e90fd44d8ce5b39c0af415e49275d1b99f30b35.zip
Starting package 1/2: mpfr:x64-osx
Building package mpfr[core]:x64-osx...
-- Using cached /Users/adam/vcpkg/downloads/mpfr-4.0.2.tar.xz
-- Extracting source /Users/adam/vcpkg/downloads/mpfr-4.0.2.tar.xz
-- Using source at /Users/adam/vcpkg/buildtrees/mpfr/src/mpfr-4-e3d076d3ce.clean
-- Configuring x64-osx-dbg
-- Configuring x64-osx-rel
-- Building x64-osx-dbg
-- Building x64-osx-rel
-- Installing: /Users/adam/vcpkg/packages/mpfr_x64-osx/share/mpfr/copyright
-- Performing post-build validation
-- Performing post-build validation done
Stored binary cache: /Users/adam/.cache/vcpkg/archives/b2/b21576f062032d14e5e5805b229a057f2c271d00.zip
Building package mpfr[core]:x64-osx... done
Installing package mpfr[core]:x64-osx...
Installing package mpfr[core]:x64-osx... done
Elapsed time for package mpfr:x64-osx: 7.758 s
Starting package 2/2: cgal:x64-osx
Building package cgal[core]:x64-osx...
-- Using cached /Users/adam/vcpkg/downloads/CGAL-cgal-v5.2.tar.gz
-- Extracting source /Users/adam/vcpkg/downloads/CGAL-cgal-v5.2.tar.gz
-- Using source at /Users/adam/vcpkg/buildtrees/cgal/src/v5.2-e7cb64511b.clean
-- Configuring x64-osx-dbg
-- Configuring x64-osx-rel
-- Building x64-osx-dbg
-- Building x64-osx-rel
-- Installing: /Users/adam/vcpkg/packages/cgal_x64-osx/share/cgal/copyright
-- Installing: /Users/adam/vcpkg/packages/cgal_x64-osx/share/cgal/usage
-- Performing post-build validation
-- Performing post-build validation done
Stored binary cache: /Users/adam/.cache/vcpkg/archives/8e/8e90fd44d8ce5b39c0af415e49275d1b99f30b35.zip
Building package cgal[core]:x64-osx... done
Installing package cgal[core]:x64-osx...
Installing package cgal[core]:x64-osx... done
Elapsed time for package cgal:x64-osx: 19.19 s

Total elapsed time: 28.8 s

The package cgal provides CMake targets:

    find_package(CGAL CONFIG REQUIRED)
    target_link_libraries(main PRIVATE CGAL::CGAL)

```

--- a/ports/cgal/CONTROL
+++ b/ports/cgal/CONTROL
@@ -1,6 +1,7 @@
 Source: cgal
 Version: 5.2
-Build-Depends: mpfr, mpir, zlib, boost-format, boost-container, boost-ptr-container, boost-iterator, boost-variant, boost-any, boost-unordered, boost-random, boost-foreach, boost-graph, boost-heap, boost-logic, boost-multiprecision, boost-interval
+Port-Version: 1
+Build-Depends: mpfr, gmp, zlib, boost-format, boost-container, boost-ptr-container, boost-iterator, boost-variant, boost-any, boost-unordered, boost-random, boost-foreach, boost-graph, boost-heap, boost-logic, boost-multiprecision, boost-interval
 Homepage: https://github.com/CGAL/cgal
 Description: The Computational Geometry Algorithms Library (CGAL) is a C++ library that aims to provide easy access to efficient and reliable algorithms in computational geometry.
 

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -1118,7 +1118,7 @@
     },
     "cgal": {
       "baseline": "5.2",
-      "port-version": 0
+      "port-version": 1
     },
     "cgicc": {
       "baseline": "3.2.19-4",

--- a/versions/c-/cgal.json
+++ b/versions/c-/cgal.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "5cbb2f2a60d864bd36098d71a050aeef43e52eff",
+      "version-string": "5.2",
+      "port-version": 1
+    },
+    {
       "git-tree": "1673f680f59b2c8c848a28c8e664b68b2ec182d7",
       "version-string": "5.2",
       "port-version": 0

